Dockerfile

```diff
@@ -0,0 +1,38 @@
+FROM fedora:27
+
+RUN dnf install -y python2 python2-junit_xml beaker-client \
+	bison flex python2-mock dnf-plugins-core
+RUN dnf builddep -y kernel
+
+# Use a default checkout of the kernel as our workdir Also make it the default directory
+# and setup an environment variable to tell the skt executable to use this workdir
+#
+# Notice that we give 777 and 666 permissions on this directory. This is because Openshift
+# likes to run containers as random UIDs, and we cannot rely on SKT being root
+RUN git clone git://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git /var/work && \
+	chmod -R ugo+w /var/work
+ENV WORKDIR=/var/work
+WORKDIR /var/work
+
+# Pull a default kernel config down for use in the container
+# It's expected that this will be overriden either by mounting a
+# new config file, or by adding one in a layered container
+RUN mkdir -p /var/config && \
+        git clone --depth=1 https://src.fedoraproject.org/git/rpms/kernel.git /tmp/kernel && \
+        mv /tmp/kernel/*.config /var/config/ && \
+        rm -rf /tmp/kernel
+ENV BASECONFIG=/var/config/kernel-x86_64.config
+
+# Setup a valid git config in the container
+RUN git --work-tree /var/work config user.email "skt@example.com" && \
+	git --work-tree /var/work config user.name "Sonic Kernel Testing"
+
+# We don't pollute the container file system  tree with
+# massive intermediate build output
+ENV KBUILD_OUTPUT=/var/build
+VOLUME [ "/var/build" ]
+
+COPY . /var/src/skt
+RUN cd /var/src/skt && ls && pip install -v .
```

Comment:
Do you think you can put everything under the same directory, so people don't have to hunt down these non-standard locations? Perhaps under "/root", where shell starts? Or, if that doesn't work, "/home/skt", or "/var/lib/skt"?

Reply:
In Openshift, the containers won't start as a consistent UID. It will be random. So /root and /home/skt will be awkward.

I'd like to /var. 
 * I wanted to use ```/var/src``` and sub-directories like ```/var/src/kernel``` for source and git repositories.
 * And ```/var/build``` and sub-directories for built output.

See #31